GA/checkpoint.py

```
@@ -0,0 +1,110 @@
+
+        Parameters
+        ----------
+        interval : int
+            The number of generations between each checkpoint save.
+        """
+
+        super().__init__()
+
+        # Validate the interval parameter
+        if interval <0:
+            raise ValueError("Checkpoint interval must be a positive integer.")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)